Error when selecting to use OAuth2

`d make whoami config`

```
? Do you want to enable sentry authentication in front of this app (effectively making the entire site private)?    No
  Yes, with HTTP Basic Authentication
> Yes, with Oauth2
  Yes, with Mutual TLS (mTLS)
Set WHOAMI_OAUTH2=true

Error: choices variable is missing.
Exiting.
make[1]: *** [Makefile:13: config-hook] Error 1
make: *** [../_scripts/Makefile.lifecycle:5: config] Error 2
```

--- a/_scripts/reconfigure_header_authorization
+++ b/_scripts/reconfigure_header_authorization
@@ -172,7 +172,7 @@ create_group() {
     ## Check if the group exists already:
     local TRAEFIK_HEADER_AUTHORIZATION_GROUPS=$(${BIN}/dotenv -f ${ENV_FILE} get TRAEFIK_HEADER_AUTHORIZATION_GROUPS)
     local INDEX=$(echo "$TRAEFIK_HEADER_AUTHORIZATION_GROUPS" | jq --compact-output 'keys | index("'${GROUP}'")')
-    if [[ $INDEX != "null" ]]; then
+    if [[ -n "$TRAEFIK_HEADER_AUTHORIZATION_GROUPS" && $INDEX != "null" ]]; then
         echo "Group ${GROUP} already exists."
         return 0
     fi

--- a/_scripts/reconfigure_header_authorization
+++ b/_scripts/reconfigure_header_authorization
@@ -171,8 +171,11 @@ create_group() {
 
     ## Check if the group exists already:
     local TRAEFIK_HEADER_AUTHORIZATION_GROUPS=$(${BIN}/dotenv -f ${ENV_FILE} get TRAEFIK_HEADER_AUTHORIZATION_GROUPS)
+    if [[ -z "$TRAEFIK_HEADER_AUTHORIZATION_GROUPS" ]]; then
+        TRAEFIK_HEADER_AUTHORIZATION_GROUPS="{}"
+    fi
     local INDEX=$(echo "$TRAEFIK_HEADER_AUTHORIZATION_GROUPS" | jq --compact-output 'keys | index("'${GROUP}'")')
-    if [[ -n "$TRAEFIK_HEADER_AUTHORIZATION_GROUPS" && $INDEX != "null" ]]; then
+    if [[ $INDEX != "null" ]]; then
         echo "Group ${GROUP} already exists."
         return 0
     fi
@@ -191,7 +194,7 @@ create_group() {
     fi
 
     ## Add the new group:
-     users=$(jq --compact-output --null-input '$ARGS.positional | unique' --args -- "${users[@]}")
+    users=$(jq --compact-output --null-input '$ARGS.positional | unique' --args -- "${users[@]}")
     local TRAEFIK_HEADER_AUTHORIZATION_GROUPS=$(echo "${TRAEFIK_HEADER_AUTHORIZATION_GROUPS}" | jq --compact-output "(.${GROUP} = ${users})")
     ${BIN}/dotenv -f ${ENV_FILE} set TRAEFIK_HEADER_AUTHORIZATION_GROUPS="${TRAEFIK_HEADER_AUTHORIZATION_GROUPS}"
     echo "Wrote new TRAEFIK_HEADER_AUTHORIZATION_GROUPS to ${ENV_FILE}"

--- a/_scripts/reconfigure_header_authorization
+++ b/_scripts/reconfigure_header_authorization
@@ -167,7 +167,17 @@ list_groups() {
 create_group() {
     set -eo pipefail
     local GROUP
-    GROUP=$(wizard ask "Enter the name of the group to create:")
+
+    while true; do
+        GROUP=$(wizard ask "Enter the name of the group to create:")
+        # Check if the input matches the allowed character set
+        if [[ "$GROUP" =~ ^[a-zA-Z0-9_]+$ ]]; then
+            break
+        else
+            echo "Invalid group name. Please use only letters, digits, or underscores."
+            echo
+        fi
+    done
 
     ## Check if the group exists already:
     local TRAEFIK_HEADER_AUTHORIZATION_GROUPS=$(${BIN}/dotenv -f ${ENV_FILE} get TRAEFIK_HEADER_AUTHORIZATION_GROUPS)

--- a/_scripts/reconfigure_oauth2
+++ b/_scripts/reconfigure_oauth2
@@ -56,4 +56,8 @@ groups=($(jq -r '. | keys[]' <<< ${TRAEFIK_HEADER_AUTHORIZATION_GROUPS}))
 ## Ask user which header authorization group to permit access to this app
 authorized_group=$(${BIN}/dotenv -f ${ENV_FILE} get ${VAR_PREFIX}_OAUTH2_AUTHORIZED_GROUP)
 
+if [ ${#groups[@]} -eq 0 ]; then
+    fault "No Traefik authorization groups exist. You need to create some authorization groups in the Traefik config first before you can select the OAuth2 option in the app."
+fi
+
 ${BIN}/reconfigure_choose ${ENV_FILE} ${VAR_PREFIX}_OAUTH2_AUTHORIZED_GROUP "Which authorization group do you want to permit access to this app?" ${groups[@]}

